signer/src/storage/in_memory.rs

```diff
@@ -810,6 +810,31 @@ impl super::DbRead for SharedStore {
 
         Ok(package)
     }
+
+    async fn get_latest_unconfirmed_sweep_transactions(
+        &self,
+        prevout_txid: &model::BitcoinTxId,
+    ) -> Result<Vec<model::SweepTransaction>, Error> {
```

Comment:
nit: wrt the pg one (and the trait docstring), this doesn't filter out confirmed ones IIUC. I guess it's fine, but maybe we should add a comment about it and revisit later?

Reply:
The
```sql
LEFT JOIN bitcoin_transactions AS btc_tx 
    ON btc_tx.txid = sweep_txs.txid
WHERE btc_tx.txid IS NULL
```
should take care of that wrt the sweep transactions themselves. If you're thinking about the `prevout_txid` which you pass in, it's expected that you as the caller have done the work to ensure that that transaction you're referring to is correct.

Reply:
Uhm I'm not sure I'm following: here I was referring to the in-memory impl not filtering out the confirmed sweep tx, while the pg does it (correctly, according to the docstring). Am I missing something?

Reply:
Ohh sorry I thought you meant with regards to the PG one. Right, i'll have to check the in-mem one or just mark it `unimplemented!()`, I think I forgot to circle back to that impl after the latest updates.